app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -86,6 +87,14 @@
     private final ResourceManager resources;
     private final D2 d2;
     private final SearchSortingValueSetter sortingValueSetter;
+    private TrackedEntityInstanceQueryCollectionRepository trackedEntityInstanceQuery;
+    private Program savedSelectedProgram;
+    private String savedTrackedEntityType;
+    private List<String> savedOrgUnits = new ArrayList<>();
+    private SortingItem savedSortingItem;
+    private List<State> savedStates = new ArrayList<>();
+    private HashMap<String, String> savedQueryData = new HashMap<>();
+    private boolean savedAssignedToMe;
```

Comment:
Do you think it makes sense to create a UI Model with a data class and simplify the comparting ? @mmmateos

Reply:
Yes, sure